build.sbt

```diff
@@ -4,24 +4,42 @@ import microsites._
 
 lazy val core = (project in file("core")).
   enablePlugins(TutPlugin, SbtOsgi).
-  settings(commonSettings, tutTargetDirectory := file(".")).
-  dependsOn(macros).
+  settings(commonSettings).
+  dependsOn(macros)
+
+// A special module for now, since tests depend on it. We should improve this organization later.
```

Comment:
The issue here is that we make each module declared with the `module` function depend on `tests` in order for them to have access to test helpers like `BaseSuite`, creating a circular dependency in the case of the `generic` module. There are a lot of things module tests have access to that they don't need (like specific test classes). 

I think that the best way to organize this is to have testkit-like code (or it can simply be code in `tests/src/main/scala`, if we don't want yet another subproject) that does not depend on `generic` so that we can have the dependencies like:

```
            +---------+
   +--------> testkit +------------------------+-------+
   |        +---------+                        |       |
   |                                           |       |
+--+---+    +---------+                    +---v---+   |
| core +----> generic +--------------------> tests |   |
+--+---+    +---------+                    +-------+   |
   |                                                   |
   |        +-------------------------+    +-----------v----------+
   +--------> other modules (compile) +----> other modules (test) |
            +-------------------------+    +----------------------+
```

I didn't comment this better because I was planning to do that (as well as other minor improvements) right after this PR was accepted, but I'll add a comment here explaining this better :)

Reply:
Regarding tut, I think that all our documentation should continue to rely on having `generic` as dependency, since that's by far the most common PureConfig use case.